### ISSUE TITLE
Pattern features do not allow adding or removing features from the pattern after initial creation (tip release)

### ISSUE THREAD

**Issue by eblanshey** [realthunder/FreeCAD#848]
I am using the latest tip release, Tip 2023.05.21

To reproduce:
1. Create a body above the XY plane with multiple features
2. Mirror the body over the XY plane, and save
3. "Set body tip" to the last feature before the mirror
4. Add a pocket to the body somewhere, save
5. Set body tip to mirror feature again
6. Edit mirror feature, add a "check" to the new pocket
7. Preview does not include the new pocket, and saving the mirror does not show the new pocket on the mirror side
8. Edit the mirror again. The new pocket is unchecked.

EDIT: the same is happening with the linear pattern as well. Can't add or remove features.

```
OS: Windows 10 Version 2009
Word size of FreeCAD: 64-bit
Version: 2023.524.0.14555 (Git shallow)
Build type: Release
Branch: HEAD
Hash: f428979aa87456d1ce57e2ea78ee9d70dc7595e4
Python 3.11.3, Qt 5.15.6, Coin 4.0.1, Vtk 9.2.5, OCC 7.7.1
Locale: English/United States (en_US)
Installed mods: 
  * AirPlaneDesign 0.4.0
  * Assembly3 0.12.0
  * Assembly4 0.50.2
  * CurvedShapes 1.0.4
  * fasteners 0.4.56
  * Help 1.0.3
  * Plot 2022.4.17
  * QuickMeasure 2022.10.28
```

**Comment by Trakyan** [comment]
I opened an issue about this a few weeks ago here
https://github.com/realthunder/FreeCAD/issues/839
It's still present in todays release but in the issue above I mentioned a work around to add/remove features to the mirror/pattern operations.

**Comment by eblanshey** [comment]
Thanks, @Trakyan. Fixed in Tip 2023.07.30.
